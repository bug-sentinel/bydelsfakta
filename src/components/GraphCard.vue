<template>
  <div class="card-container" :class="{ large: settings.size === 'large' }">
    <div class="card">
      <div class="tabs" role="tablist">
        <button
          role="tab"
          :aria-selected="{ true: active === index }"
          :aria-label="tab.label"
          v-for="(tab, index) in settings.tabs"
          :key="index"
          @click="activeTab(index)"
          :class="{ active: active === index }"
          class="tabs__button"
          v-text="tab.label"
        ></button>
        <div
          @keydown.escape="closeMenu()"
          v-click-outside="closeMenu"
          class="context-menu"
          role="menu"
        >
          <button
            class="context-menu__button"
            @click="showDropdown = !showDropdown"
            aria-haspopup="true"
            :aria-label="showDropdown ? 'Lukk meny' : 'Ã¥pne meny'"
          >
            <i class="material-icons">{{showDropdown ? 'close' : 'menu'}}</i>
          </button>
          <div v-if="showDropdown" class="context-menu__dropdown">
            <button
              @click="savePng(settings.tabs[active].id)"
              @keyup.enter="saveSvg(settings.tabs[active].id)"
              class="context-menu__dropdown-item"
              tabindex="0"
              aria-label="lagre graf som png"
            >
              <i class="material-icons context-menu__dropdown-item-icon">photo_size_select_actual</i>
              <span>Last ned som PNG</span>
            </button>
            <button
              class="context-menu__dropdown-item"
              aria-label="lagre graf som svg"
              tabindex="0"
              @click="saveSvg(settings.tabs[active].id)"
              @keyup.enter="saveSvg(settings.tabs[active].id)"
            >
              <i class="material-icons context-menu__dropdown-item-icon">photo_size_select_actual</i>
              <span>Lagre som SVG</span>
            </button>
          </div>
        </div>
      </div>
      <graph
        v-if="settings.tabs[active] !== undefined"
        :settings="settings.tabs[active]"
        ref="graph"
      />
    </div>
  </div>
</template>

<script>
import { saveSvgAsPng } from 'save-svg-as-png';
import Graph from './Graph.vue';
import downloadSvg from '../util/downloadSvg';

export default {
  name: 'GraphCard',
  components: { Graph },
  data() {
    return {
      active: 0,
      showDropdown: false,
    };
  },
  props: {
    settings: {
      type: Object,
      required: true,
    },
  },
  methods: {
    closeMenu() {
      if (this.showDropdown) {
        this.showDropdown = false;
      }
    },

    activeTab(index) {
      this.active = index;
    },

    saveSvg(id) {
      const filename = `${this.$route.params.bydel}_${id}.svg`;
      const svgData = this.$refs.graph.$refs.svg.outerHTML;
      downloadSvg(svgData, filename);
    },

    savePng(id) {
      const file = this.$refs.graph.$refs.svg;
      const filename = `${this.$route.params.bydel}_${id}.png`;
      saveSvgAsPng(file, filename, {
        backgroundColor: 'white',
        encoderOptions: 1,
        width: file.width.baseVal.value + 40,
        height: file.height.baseVal.value + 40,
        top: -20,
        left: -20,
      });
    },
  },
};
</script>

<style scoped lang="scss">
@import './../styles/colors';

.card-container {
  padding: 0.5em;
  flex: 50% 1 10;
  min-width: 300px;
  overflow: hidden;

  @media screen and (min-width: 950px) {
    min-width: 400px;
    max-width: 50%;
  }

  &.large {
    max-width: 100%;
    flex-basis: 100%;
  }
}

.card {
  background: white;
  border: 1px solid $color-grey-100;
  border-radius: 3px;
  box-shadow: 0 1.5px 3px rgba($color-grey-600, 0.95);
  width: 100%;
}

/* Style the tabs */
.tabs {
  background-color: $color-grey-50;
  border-bottom: 1px solid $color-grey-100;
  border-top-left-radius: 4px;
  border-top-right-radius: 4px;
  display: flex;
  // justify-content: space-between;
  width: 100%;

  div:first-of-type a:first-child {
    border-top-left-radius: 3px;
  }

  &__button {
    background-color: $color-grey-50;
    color: rgb(41, 40, 88);
    cursor: pointer;
    float: left;
    font-weight: bold;
    transition: background-color 0.2s;
    padding: 14px 24px;

    &.active {
      background-color: #fff;
      box-shadow: 0 0 0 1px #e0e0e0;
      cursor: default;
      position: relative;

      &:after {
        background-color: white;
        bottom: -2px;
        content: '';
        height: 2px;
        left: 0;
        position: absolute;
        right: 0;
      }
    }

    /* Change background color of tabs on hover */
    &:hover:not(.active) {
      background-color: #efefef;
      box-shadow: 0 0 0 1px #e0e0e0;
      position: relative;
    }
  }
}

.context-menu {
  position: relative;
  margin-left: auto;

  &__button {
    background-color: $color-light-blue-2;
    border-top-right-radius: 3px;
    height: 3.5rem;
    width: 4rem;
  }

  &__dropdown {
    background-color: rgb(178, 210, 216);
    position: absolute;
    right: 0;
    width: 200px;
    z-index: 1;

    &-item {
      box-shadow: 0 2px 2px 0 $color-grey-200;
      color: $color-purple;
      display: flex;
      flex-direction: row;
      font-weight: bold;
      padding: 0.5rem;
      width: 100%;

      &:hover {
        background-color: lighten($color-light-blue-2, 5%);
      }

      &-icon {
        color: rgb(41, 40, 88);
        font-size: 16px;
      }

      span {
        margin-left: 0.5rem;
      }
    }
  }
}
</style>
