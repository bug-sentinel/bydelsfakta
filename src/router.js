import Vue from 'vue';
import Router from 'vue-router';
import Home from './views/Home.vue';
import Bydel from './views/Bydel.vue';
import Alder from './views/subviews/Alder.vue';
import Boligpriser from './views/subviews/Boligpriser.vue';
import Bygningstyper from './views/subviews/Bygningstyper.vue';
import Eierform from './views/subviews/Eierform.vue';
import Flytting from './views/subviews/Flytting.vue';
import Folkemengde from './views/subviews/Folkemengde.vue';
import Husholdninger from './views/subviews/Husholdninger.vue';
import Innvandrerbefolkningen from './views/subviews/Innvandrerbefolkningen.vue';
import Kommunaleboliger from './views/subviews/Kommunaleboliger.vue';
import Levekaar from './views/subviews/Levekaar.vue';
import Trangboddhet from './views/subviews/Trangboddhet.vue';

import Graph from './components/Graph.vue';

Vue.use(Router);

export default new Router({
  routes: [
    {
      path: '/',
      name: 'Home',
      component: Home,
    },
    {
      path: '/graphA',
      name: 'Graph1',
      props: { settings: { url: '/mockDataA.json', template: 'a' } },
      component: Graph,
    },
    {
      path: '/graphB',
      name: 'Graph2',
      props: { settings: { url: '/mockDataA2.json', template: 'a' } },
      component: Graph,
    },
    {
      path: '/graphC',
      name: 'Graph3',
      props: { settings: { url: '/mockDataB.json', template: 'b' } },
      component: Graph,
    },
    {
      path: '/bydel/:bydel',
      name: 'Bydel',
      component: Bydel,
    },
    {
      path: '/bydel/:bydel/folkemengde',
      name: 'BydelFolkemengde',
      component: Folkemengde,
    },
    {
      path: '/bydel/:bydel/alder',
      name: 'BydelAlder',
      component: Alder,
    },
    {
      path: '/bydel/:bydel/innvandrerbefolkningen',
      name: 'BydelInnvandrerbefolkningen',
      component: Innvandrerbefolkningen,
    },
    {
      path: '/bydel/:bydel/husholdninger',
      name: 'BydelHusholdninger',
      component: Husholdninger,
    },
    {
      path: '/bydel/:bydel/trangboddhet',
      name: 'BydelTrandboddhet',
      component: Trangboddhet,
    },
    {
      path: '/bydel/:bydel/eierform',
      name: 'BydelEierform',
      component: Eierform,
    },
    {
      path: '/bydel/:bydel/kommunaleboliger',
      name: 'BydelKommunaleboliger',
      component: Kommunaleboliger,
    },
    {
      path: '/bydel/:bydel/flytting',
      name: 'BydelFlytting',
      component: Flytting,
    },
    {
      path: '/bydel/:bydel/bygningstyper',
      name: 'BydelBygningstyper',
      component: Bygningstyper,
    },
    {
      path: '/bydel/:bydel/boligpriser',
      name: 'BydelBoligpriser',
      component: Boligpriser,
    },
    {
      path: '/bydel/:bydel/levekaar',
      name: 'Bydellevekaar',
      component: Levekaar,
    },
    {
      path: '/sammenlign/:bydel*',
      name: 'Sammenlign',
      component: Bydel,
      children: [
        {
          path: '/folkemengde',
          name: 'Folkemengde',
          component: Folkemengde,
        },
        {
          path: '/alder',
          name: 'Alder',
          component: Alder,
        },
        {
          path: '/innvandrerbefolkningen',
          name: 'Innvandrerbefolkningen',
          component: Innvandrerbefolkningen,
        },
        {
          path: '/husholdninger',
          name: 'Husholdninger',
          component: Husholdninger,
        },
        {
          path: '/trangboddhet',
          name: 'Trandboddhet',
          component: Trangboddhet,
        },
        {
          path: '/eierform',
          name: 'Eierform',
          component: Eierform,
        },
        {
          path: '/kommunaleboliger',
          name: 'Kommunaleboliger',
          component: Kommunaleboliger,
        },
        {
          path: '/flytting',
          name: 'Flytting',
          component: Flytting,
        },
        {
          path: '/bygningstyper',
          name: 'Bygningstyper',
          component: Bygningstyper,
        },
        {
          path: '/boligpriser',
          name: 'Boligpriser',
          component: Boligpriser,
        },
        {
          path: '/levekaar',
          name: 'levekaar',
          component: Levekaar,
        },
      ],
    },
  ],
});
