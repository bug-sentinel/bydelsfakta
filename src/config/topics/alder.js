import { apiUrl, baseUrl } from '../../util/config';

const API = `${apiUrl}/api/dataset`;

export default {
  text: 'Alder',
  value: 'alder',
  production: true,
  cards: [
    {
      size: 'large',
      heading: 'Befolkningen etter alder',
      map: false,
      production: true,
      tabs: [
        {
          active: false,
          label: 'Antall',
          id: 'alder_segment_antall',
          template: 'd',
          heading: 'Aldersdistribusjon (antall)',
          url: `${API}/alder-distribusjon-status`,
          method: 'value',
          help: 'Dette er hjelp',
          production: true,
        },
        {
          active: false,
          label: 'Andel',
          id: 'alder_segment_andel',
          template: 'd',
          heading: 'Aldersdistribusjon (andel)',
          url: `${API}/alder-distribusjon-status`,
          method: 'ratio',
          production: true,
        },
      ],
    },
    {
      size: 'large',
      heading: 'Befolkningspyramide',
      map: false,
      production: true,
      tabs: [
        {
          active: false,
          label: 'Status',
          id: 'alder_distribusjon_status',
          template: 'e',
          heading: 'Aldersdistribusjon fordelt på kjønn',
          method: 'value',
          url: `${API}/alder-distribusjon-status`,
          production: true,
        },
      ],
    },
    {
      size: 'large',
      heading: 'Gjennomsnitt- og medianalder',
      production: true,
      map: {
        labels: ['Lavere gjennomsnittsalder', 'Høyere gjennomsnittsalder'],
        scale: [32, 43],
        reverse: true,
        method: 'avg',
        url: `${API}/alder-distribusjon-status`,
        production: true,
      },
      tabs: [
        {
          active: false,
          label: 'Status',
          id: 'alder_medianalder_status',
          method: 'total',
          heading: 'Gjennomsnittsaldre',
          template: 'f',
          help:
            'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Mauris vestibulum malesuada arcu id pulvinar. Donec congue quam nec elit molestie, vel feugiat lorem euismod. Pellentesque habitant morbi tristique senectus et netus et malesuada fames ac turpis egestas. Integer feugiat urna risus, nec interdum risus laoreet id.',
          url: `${API}/alder-distribusjon-status`,
          production: true,
        },
      ],
    },
  ],
  options: {
    kategori: 'Befolkning',
    tema: 'Alder',
    bgImage: `${baseUrl}/img/alder`,
    txtColor: 'rgb(245, 173, 165)',
  },
  related: ['folkemengde', 'husholdninger', 'levekaar'],
};
