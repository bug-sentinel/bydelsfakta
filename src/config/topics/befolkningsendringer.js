import { baseUrl } from '../../util/config';

// const API = `${apiUrl}/api/dataset`;

export default {
  text: 'Befolkningsendringer',
  value: 'befolkningsendringer',
  production: false,
  cards: [
    {
      size: 'large',
      heading: 'Flytting etter alder',
      production: false,
      tabs: [
        {
          label: 'Status',
          method: 'value',
          heading: 'Flytting etter alder og innvandringskategori',
          template: 'm',
          url: `${baseUrl}/mockDataFlytting.json`,
          production: true,
        },
      ],
    },
  ],
  options: {
    kategori: 'Befolkning',
    tema: 'Befolknings&shy;endringer',
    bgImage: `${baseUrl}/img/folkemengde`,
    txtColor: 'rgb(245, 173, 165)',
  },
  related: ['alder', 'folkemengde', 'innvandrerbefolkningen'],
};
