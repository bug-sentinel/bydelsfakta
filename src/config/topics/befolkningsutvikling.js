import { apiUrl, baseUrl } from '../../util/config';
import source from './dataSources';

const API = `${apiUrl}/api/dataset`;

export default {
  text: 'Befolkningsutvikling',
  value: 'befolkningsutvikling',
  cards: [
    {
      size: 'large',
      heading: 'Befolkningsutvikling',
      about: {
        info: 'Statistikken viser folkemengden per 1.1. hvert år. For flere befolkningstabeller se under «Befolkning» i Oslo kommunes statistikkbank.',
        sources: [source.ssb],
        externalInfo:
          'http://statistikkbanken.oslo.kommune.no/webview/index.jsp?catalog=http%3A%2F%2Fstatistikkbanken.oslo.kommune.no%3A80%2Fobj%2FfCatalog%2FCatalog48&submode=catalog&mode=documentation&top=yes',
      },
      tabs: [
        {
          label: 'Folkemengde',
          heading: 'Befolkningsutvikling',
          id: 'befolkningsutvikling_antall',
          method: 'value',
          template: 'lines',
          url: `${API}/folkemengde-utvikling-historisk`,
        },

        {
          heading: 'Prosentvis årlig utvikling i folkemengde',
          label: 'Prosentvis endring',
          id: 'befolkningsutvikling_andel',
          method: 'ratio',
          template: 'lines',
          url: `${API}/folkemengde-utvikling-historisk-prosent`,
        },
      ],
    },
    {
      size: 'large',
      heading: 'Befolkningsvekst',
      about: {
        info: 'Statistikken viser folkemengden per 1.1. hvert år. For flere befolkningstabeller se under «Befolkning» i Oslo kommunes statistikkbank.',
        sources: [source.ssb],
        externalInfo:
          'http://statistikkbanken.oslo.kommune.no/webview/index.jsp?catalog=http%3A%2F%2Fstatistikkbanken.oslo.kommune.no%3A80%2Fobj%2FfCatalog%2FCatalog48&submode=catalog&mode=documentation&top=yes',
      },
      tabs: [
        {
          label: 'Status',
          heading: 'Befolkningsvekst',
          id: 'befolkningsvekst',
          url: `${API}/nokkeltall-om-befolkningen`,
          template: 'populationDetailsTable',
        },
      ],
    },
  ],
  related: ['alder', 'innvandrerbefolkningen', 'fodte-dode-flytting'],
  options: {
    kategori: 'Befolkning',
    tema: 'Befolkningsutvikling',
    bgImage: `${baseUrl}/img/folkemengde`,
    txtColor: 'rgb(245, 173, 165)',
  },
};
