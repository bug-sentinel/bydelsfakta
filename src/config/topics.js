import alder from './topics/alder';
import fodteDodeFlytting from './topics/fodteDodeFlytting';
import boligpriser from './topics/boligpriser';
import bygningstyper from './topics/bygningstyper';
import eierform from './topics/eierform';
import befolkningsutvikling from './topics/befolkningsutvikling';
import husholdninger from './topics/husholdninger';
import innvandrerbefolkningen from './topics/innvandrerbefolkningen';
import levekaar from './topics/levekaar';
import romPerPerson from './topics/romPerPerson';

export const topicNames = [
  'befolkningsutvikling',
  'fodte-dode-flytting',
  'alder',
  'innvandrerbefolkningen',
  'husholdninger',
  'levekaar',
  'eierform',
  'bygningstyper',
  'boligpriser',
  'rom-per-person',
];

export const disabledTopics = [];

export const categories = [
  {
    kategori: 'Befolkning',
    color: 'rgb(182, 63, 50)',
    links: ['befolkningsutvikling', 'fodte-dode-flytting', 'alder', 'innvandrerbefolkningen', 'husholdninger'],
  },
  {
    kategori: 'Boforhold',
    color: 'rgb(27, 173, 120)',
    links: ['rom-per-person', 'eierform', 'bygningstyper', 'boligpriser'],
  },
  {
    kategori: 'Levek√•r',
    color: 'rgb(219, 160, 52)',
    links: ['levekaar'],
  },
];

export const topics = {
  alder,
  'fodte-dode-flytting': fodteDodeFlytting,
  boligpriser,
  bygningstyper,
  eierform,
  befolkningsutvikling,
  husholdninger,
  innvandrerbefolkningen,
  levekaar,
  'rom-per-person': romPerPerson,
};
