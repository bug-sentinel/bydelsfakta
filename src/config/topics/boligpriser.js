import { apiUrl, baseUrl } from '../../util/config';
import source from './dataSources';

const API = `${apiUrl}/api/dataset`;

export default {
  text: 'Boligpriser',
  value: 'boligpriser',
  cards: [
    {
      size: 'large',
      heading: 'Boligpris for blokkleiligheter',
      about: {
        info: '<p>Statistikken viser gjennomsnittlig kvadratmeterpris på omsatte boliger* via FINN.no i tillegg til salg registrert i grunnboken. Pris er kjøpesummen oppgitt i kjøpekontrakten. For borettslagsboliger er både innskudd og andel av fellesgjeld inkludert i prisen. Arealbegrepet som benyttes i statistikken er P-ROM. Ettersom små boliger gjerne har høyere kvadratmeterpris enn store boliger, er det kun blokkleiligheter som benyttes i dette materialet. For delbydeler med få omsatte leiligheter er dette et mindre robust mål enn der hvor det er mange omsatte leiligheter. Kvadratmeterpriser for 2018 og nyere er produsert av Virdi AS.</p> <p>*Omsatte boliger inkluderer:</p> <ol><li>Alle salg via FINN.no, og</li><li>Alle salg via grunnboken som:</li><li>står oppført som et fritt salg (i motsetning til arv e.l.)</li><li>hvor kjøper og selger ikke er samme person</li><li>salget er fullstendig (100% av eiendommen er solgt)</li><li>det er kun en enhet som er solgt (ikke en boligblokk e.l.)</li></ol><p>Kvadratmeterpriser for årene 2004-2017 er produsert av SSB. Denne statistikken er basert på finn.no sitt prisregister over omsatte boliger. I tillegg supplerer SSB boligomsetningene med data fra Matrikkelen.Statistikken for 2018 og nyere er produsert av Virdi AS. Dette medfører et brudd i tidsserien. Vi anser imidlertid sammenlignbarheten mellom de to statistikkene som temmelig god. Begge produsentene henter salgsdata fra FINN.no, men Virdi AS supplerer datagrunnlaget med salg fra grunnboken og har i så måte et litt større datagrunnlag enn SSB. I statistikkbanken til Oslo kommune er det tilgjengeliggjort statistikk fra både SSB og Virdi AS for årene 2018 og 2019. Gjennomsnittlig kvadratmeterpris i de to tabellene for de to overlappende årene, er ikke identiske, men avvikene er små og går i begge retninger.</p>',
        sources: [source.ssb],
        externalInfo:
          'http://statistikkbanken.oslo.kommune.no/webview/index.jsp?Geografisubset=30100%2C30101+-+30117&headers=r&stubs=Geografi&measure=common&virtualslice=Gjennomsnittligkvadratmeterpris_value&layers=Boligtype&layers=virtual&study=http%3A%2F%2F192.168.101.44%3A80%2Fobj%2FfStudy%2FNPS-omsatteboliger-kvadratmeterpris&Boligtypesubset=1&mode=cube&v=2&virtualsubset=Gjennomsnittligkvadratmeterpris_value&rsubset=2004+-+2018&Boligtypeslice=1&measuretype=4&cube=http%3A%2F%2F192.168.101.44%3A80%2Fobj%2FfCube%2FNPS-omsatteboliger-kvadratmeterpris_C1&top=yes',
      },
      map: {
        labels: ['Lavere boligpriser', 'Høyere boligpriser'],
        heading: 'Gjennomsnittlig kvadratmeterpris blokkleiligheter',
        method: 'value',
        url: `${API}/boligpriser-blokkleiligheter-status`,
      },
      tabs: [
        {
          label: 'Status',
          id: 'boligpriser_status',
          heading: 'Gjennomsnittlig kvadratmeterpris for blokkleiligheter',
          method: 'value',
          template: 'bars',
          url: `${API}/boligpriser-blokkleiligheter-status`,
        },
        {
          label: 'Historisk',
          id: 'boligpriser_historisk',
          heading: 'Gjennomsnittlig kvadratmeterpris for blokkleiligheter',
          method: 'value',
          template: 'lines',
          url: `${API}/boligpriser-blokkleiligheter-historisk`,
        },
      ],
    },
    {
      size: 'large',
      heading: 'Median totalpris',
      about: {
        info: '<p>Statistikken viser median totalpris på omsatte boliger via FINN.no i tillegg til salg registrert i grunnboken. Pris er kjøpesummen oppgitt i kjøpekontrakten. For borettslagsboliger er både innskudd og andel av fellesgjeld inkludert i prisen.</p><p>Omsatte boliger inkluderer:</p><ol><li>Alle salg via FINN.no, og</li><li>Alle salg via grunnboken som:<ul><li>står oppført som et fritt salg (i motsetning til arv e.l.)</li><li>hvor kjøper og selger ikke er samme person</li><li>salget er fullstendig (100% av eiendommen er solgt)</li><li>det er kun en enhet som er solgt (ikke en boligblokk e.l.)</ul></ol><p>Totalpris = Salgspris + fellesgjeld<br>Median totalpris = Totalpris på boligen som er midt i rekken hvis salgene sorteres etter pris.</p>',
        sources: [source.virdi],
      },
      map: {
        labels: ['Lavere boligpriser', 'Høyere boligpriser'],
        heading: 'Median totalpris',
        method: 'value',
        url: `${API}/median-totalpris-status`,
      },
      tabs: [
        {
          label: 'Status',
          id: 'median_totalpris_status',
          heading: 'Median totalpris for boliger',
          method: 'value',
          template: 'bars',
          url: `${API}/median-totalpris-status`,
        },
        {
          label: 'Historisk',
          id: 'median_totalpris_historisk',
          heading: 'Median totalpris for boliger',
          method: 'value',
          template: 'lines',
          url: `${API}/median-totalpris-historisk`,
        },
      ],
    },
  ],
  options: {
    kategori: 'Boforhold',
    tema: 'Boligpriser',
    bgImage: `${baseUrl}/img/boligpriser`,
    txtColor: 'rgb(199, 247, 201)',
  },
  related: ['rom-per-person', 'eierform', 'fodte-dode-flytting'],
};
