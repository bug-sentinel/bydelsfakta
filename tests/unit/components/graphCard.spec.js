import { mount, createLocalVue } from '@vue/test-utils';
import vueResize from 'vue-resize';
import router from '../../../src/router';
import GraphCard from '../../../src/components/GraphCard.vue';
import clickOutside from '../../../src/directives/clickOutside';
import topics from '../../../src/config/topics';

describe('GraphCard', () => {
  let wrapper = null;

  beforeEach(() => {
    const localVue = createLocalVue();
    localVue.use(vueResize);
    localVue.use(router);
    localVue.directive('click-outside', clickOutside);
    wrapper = mount(GraphCard, {
      propsData: {
        settings: topics['alder'].subpages[0].graphs[0],
      },
      localVue,
      router,
    });
  });

  afterEach(() => {
    wrapper.destroy();
  });

  test('renders graphCard-component and finds card-container-class', () => {
    expect(wrapper.classes('card-container')).toBe(true);
  });

  test('renders correctly', () => {
    expect(wrapper.element).toMatchSnapshot();
  });

  test('change an active tab', () => {
    wrapper.setProps({ settings: topics['eierform'].subpages[0].graphs[0] });
    wrapper.vm.activeTab(1);
    expect(wrapper.vm.active).toEqual(1);
    expect(wrapper.element).toMatchSnapshot();
  });

  test('change active tab multiple times', () => {
    wrapper.setProps({ settings: topics['husholdninger'].subpages[0].graphs[0] });
    expect(wrapper.vm.active).toEqual(0);
    wrapper.vm.activeTab(1);
    expect(wrapper.vm.active).toEqual(1);
    wrapper.vm.activeTab(2);
    expect(wrapper.vm.active).toEqual(2);
  });

  test('if dropdown is false, stay false', () => {
    expect(wrapper.vm.showDropdown).toEqual(false);
    wrapper.vm.closeMenu();
    expect(wrapper.vm.showDropdown).toEqual(false);
  });

  test('if dropdown is true, change to false', () => {
    wrapper.setData({ showDropdown: true });
    expect(wrapper.vm.showDropdown).toEqual(true);
    wrapper.vm.closeMenu();
    expect(wrapper.vm.showDropdown).toEqual(false);
  });

  test('saveSvg is called once', () => {
    const spy = jest.spyOn(wrapper.vm, 'saveSvg');
    spy(wrapper.vm.settings.tabs[wrapper.vm.active].id);
    expect(spy).toHaveBeenCalledTimes(1);
  });
});
