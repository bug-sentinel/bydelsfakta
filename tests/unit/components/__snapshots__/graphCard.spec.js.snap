// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GraphCard change an active tab 1`] = `
<div
  class="card-container large"
>
  <div
    class="card"
  >
    <header
      class="card__header"
    >
      <div
        class="card__headertext"
      >
        <h2
          class="card__title"
        >
          
        </h2>
         
        <span
          class="card__published"
        >
          Oppdatert 01.09.2018
        </span>
      </div>
       
      <nav
        class="card__nav"
      >
        <div
          class="tabs"
          role="tablist"
        >
          <button
            aria-label="Status"
            aria-selected="[object Object]"
            class="tabs__tab"
            id="tabButton-0"
            role="tab"
          >
            Status
          </button>
          <button
            aria-label="Matrise"
            aria-selected="[object Object]"
            class="tabs__tab active"
            id="tabButton-1"
            role="tab"
          >
            Matrise
          </button>
          <button
            aria-label="Historisk"
            aria-selected="[object Object]"
            class="tabs__tab"
            id="tabButton-2"
            role="tab"
          >
            Historisk
          </button>
           
          <div
            class="card__toggle-menu"
          >
            <button
              class="card__toggle-button card__toggle-button--active"
            >
              <i
                aria-hidden="true"
                class="material-icons context-menu__dropdown-item-icon"
              >
                bar_chart
              </i>
            </button>
             
            <button
              class="card__toggle-button"
            >
              <i
                aria-hidden="true"
                class="material-icons context-menu__dropdown-item-icon"
              >
                table_chart
              </i>
            </button>
             
            <button
              class="card__toggle-button"
            >
              <i
                aria-hidden="true"
                class="material-icons context-menu__dropdown-item-icon"
              >
                map
              </i>
            </button>
          </div>
        </div>
         
        <div
          class="context-menu"
          role="menu"
        >
          <button
            aria-haspopup="true"
            aria-label="Åpne meny"
            class="context-menu__button"
            id="context-menu-button"
          >
            <i
              aria-hidden="true"
              class="material-icons"
            >
              menu
            </i>
             
            <span>
              Valg
            </span>
          </button>
           
          <!---->
        </div>
      </nav>
    </header>
     
    <div
      class="graph__shadow"
      role="tabpanel"
    >
      <div
        aria-hidden="true"
        class="graph__container"
      >
        <div
          class="spinner"
        >
          <img
            alt=""
            src="../assets/spinner.svg"
          />
        </div>
         
        <!---->
         
        <svg
          aria-hidden="true"
          class="graph__svg loading"
        />
      </div>
       
      <div
        class="graph__tablecontainer visually-hidden"
      >
        <table>
          <caption />
           
          <thead />
           
          <tbody />
        </table>
      </div>
       
      <div
        class="resize-observer"
        data-v-b329ee4c=""
        tabindex="-1"
      >
        <object
          aria-hidden="true"
          data="about:blank"
          tabindex="-1"
          type="text/html"
        />
      </div>
    </div>
     
    <!---->
  </div>
</div>
`;

exports[`GraphCard renders correctly 1`] = `
<div
  class="card-container large"
>
  <div
    class="card"
  >
    <header
      class="card__header"
    >
      <div
        class="card__headertext"
      >
        <h2
          class="card__title"
        >
          
        </h2>
         
        <span
          class="card__published"
        >
          Oppdatert 01.09.2018
        </span>
      </div>
       
      <nav
        class="card__nav"
      >
        <div
          class="tabs"
          role="tablist"
        >
          <button
            aria-label="Antall"
            aria-selected="[object Object]"
            class="tabs__tab active"
            id="tabButton-0"
            role="tab"
          >
            Antall
          </button>
          <button
            aria-label="Andel"
            aria-selected="[object Object]"
            class="tabs__tab"
            id="tabButton-1"
            role="tab"
          >
            Andel
          </button>
           
          <div
            class="card__toggle-menu"
          >
            <button
              class="card__toggle-button card__toggle-button--active"
            >
              <i
                aria-hidden="true"
                class="material-icons context-menu__dropdown-item-icon"
              >
                bar_chart
              </i>
            </button>
             
            <button
              class="card__toggle-button"
            >
              <i
                aria-hidden="true"
                class="material-icons context-menu__dropdown-item-icon"
              >
                table_chart
              </i>
            </button>
             
            <button
              class="card__toggle-button"
            >
              <i
                aria-hidden="true"
                class="material-icons context-menu__dropdown-item-icon"
              >
                map
              </i>
            </button>
          </div>
        </div>
         
        <div
          class="context-menu"
          role="menu"
        >
          <button
            aria-haspopup="true"
            aria-label="Åpne meny"
            class="context-menu__button"
            id="context-menu-button"
          >
            <i
              aria-hidden="true"
              class="material-icons"
            >
              menu
            </i>
             
            <span>
              Valg
            </span>
          </button>
           
          <!---->
        </div>
      </nav>
    </header>
     
    <div
      class="graph__shadow"
      role="tabpanel"
    >
      <div
        aria-hidden="true"
        class="graph__container"
      >
        <div
          class="spinner"
        >
          <img
            alt=""
            src="../assets/spinner.svg"
          />
        </div>
         
        <!---->
         
        <svg
          aria-hidden="true"
          class="graph__svg loading"
        />
      </div>
       
      <div
        class="graph__tablecontainer visually-hidden"
      >
        <table>
          <caption />
           
          <thead />
           
          <tbody />
        </table>
      </div>
       
      <div
        class="resize-observer"
        data-v-b329ee4c=""
        tabindex="-1"
      >
        <object
          aria-hidden="true"
          data="about:blank"
          tabindex="-1"
          type="text/html"
        />
      </div>
    </div>
     
    <!---->
  </div>
</div>
`;
